tee-worker/omni-executor/intent/executors/cross-chain/src/cross_chain.rs

```diff
@@ -97,20 +119,72 @@ impl<BinanceClient: BinanceApi, SolanaClient: SolanaClientTrait>
 		.await?;
 
 		// TODO: this should be abstracted away
-		let (payout_amount, payout_amount_u256) = self
-			.do_binance_swap(
-				omni_account,
-				swap_order.from_asset.clone(),
-				from_amount,
-				from_address,
-				pumpx_config.wallet_index,
-				false,
-			)
-			.await?;
+		// todo we only transfer to binance without swap
+
+		let should_wait_for_deposit_confirm = false;
+
+		self.do_binance_deposit(
+			omni_account,
+			swap_order.from_asset.clone(),
+			from_amount_decimal,
+			from_address.clone(),
+			pumpx_config.wallet_index,
+			should_wait_for_deposit_confirm,
+		)
+		.await?;
+
+		if instant {
```

Comment:
Why do we only lock and release assets (before and after binance deposit) if `instant` ?

Reply:
For instant flow we do not trade assets and withdraw them from binance to user's wallet. 
Once they are deposited they are controlled by binance account owner -> so lock can be released.

Reply:
Yeah I got that. My question is, do we need to lock assets if the flow is not instant?

Reply:
For non instant flow where assets needs to be deposited -> traded -> withdrawn there is no way to spent more than currently owning.

For instant flow accounting contract is used as a source account for single chain swap so user could submit multiple swap intents and total amount traded could eventually exceed his balance.

Reply:
Currently we do not need to lock assets for non instant flow.